src/release_notes_workflow/simple_changelog_processor.py

```diff
@@ -0,0 +1,692 @@
+    * Other: If the commit cannot be clearly categorized into any of the above categories through message analysis, place it in "Other"
+   - Do not lose any commit information, even if it doesn't match any category
+
+4. **Entry Format:**
+   - Use PullRequestSubject as the main content for each entry
+   - Extract PR number from PullRequestSubject (format: (#123))
+   - Format: `* <description> ([#<number>]({repository_url}/pull/<number>))`
+   - Always use asterisk (*) for bullet points
+   - Always wrap PR links in parentheses
+   - **IMPORTANT**: Capitalize the first letter of every entry description (after the asterisk and space)
+
+5. **Output Requirements:**
+   - Generate markdown with ### headers for each category
+   - Only include categories that have entries
+   - Sort categories in this order: Breaking Changes, Features, Enhancements, Bug Fixes, Infrastructure, Documentation, Maintenance, Refactoring, Other
+   - Each entry should be a single line with proper PR link formatting
+
+6. **PR Link Format:**
+   - Extract PR number from PullRequestSubject
+   - Format as: `([#<number>]({repository_url}/pull/<number>))`
+   - Example: `([#456](https://github.com/opensearch-project/anomaly-detection/pull/456))`
+   
+7. **Important Notes:**
+   - Every commit should be categorized into exactly one category
+   - If you cannot determine the appropriate category from labels OR content analysis, place the entry in an "Unknown" category
+   - Do not skip any commits - every entry must appear somewhere in the release notes
+   - Prioritize Message over PullRequestSubject for determining category when using fallback analysis
+
+Generate the release notes in proper OpenSearch format:
+Generate the release notes in category-first format like this example:
+
+## FEATURES
+* Feature description ([#123](https://github.com/opensearch-project/alerting/pull/123))
+* Another feature description ([#124](https://github.com/opensearch-project/security/pull/124))
+
+## ENHANCEMENTS
+* Enhancement description ([#125](https://github.com/opensearch-project/ml-commons/pull/125))
+* Another enhancement description ([#126](https://github.com/opensearch-project/neural-search/pull/126))"""
+
+    body = {
+        "anthropic_version": "bedrock-2023-05-31",
+        "max_tokens": 8000,
+        "messages": [{"role": "user", "content": prompt}],
+        "temperature": 0
+    }
+    
+    response = bedrock.invoke_model(
+        modelId='us.anthropic.claude-3-7-sonnet-20250219-v1:0',
+        body=json.dumps(body),
+        contentType='application/json'
+    )
+    
+    result = json.loads(response['body'].read())
+    return result['content'][0]['text']
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Simple OpenSearch changelog processor")
+    parser.add_argument("--token", help="GitHub token")
```

Comment:
Search for GITHUB_TOKEN in the env variables rather than providing it as an input. Also it should not be needed for reading files. Checkout [release_notes](https://github.com/opensearch-project/opensearch-build/tree/7a62a342418a59b2da47da6d1909177001bf7502/src/release_notes_workflow) on existing approach.